### PR TITLE
Ruby on Rails capstone project - Budget app

### PR DESCRIPTION
**Pull request by ntzwilly**
### In this project, I implemented the pages below: 

- Splash screen
  - A simple page with the name of the app and links to the sign up and log in pages.
- Sign up and login pages
  - The user can register in the app with full name, email, and password.
  - The user can log into the app using email and password.
  - If the user is not logged in, they can't access pages that require the user to be logged in.
- Home page (categories page)
  - When the user logs in, they are presented with the categories page.
  - For each category, the user can see their name, icon and the total amount of all the transactions that belongs to that category.
  - When the user clicks (or taps) on a category item, the application navigates to the transactions page for that category.
  - There is a button "add a new category" at the bottom that brings the user to the page to create a new category.
- Transactions page
  - For a given category, the list of transactions is presented, ordered by the most recent.
  - At the top of the page the user could see the total amount for the category (sum of all of the amounts of the transactions in that category).
  - There is a button "add a new transaction" at the bottom that brings the user to the page to create a new transaction.
  - When the user clicks on the "Back" button (<), the user navigates to the home page.
- "Add a new category" page
  - The user fills out a form to create a new category, indicating their name and icon (both mandatory).
  - The user click (or taps) the "Save" button to create the new category, and is taken to the home page on success.
  - When the user clicks on the "Back" button (<), the user navigates to the home page.
- "Add a new transaction" page
  - The user fills out a form to create a new transaction with:
      -  name
      -  amount 
      -  categories 
  - The user click (or taps) the "Save" button to create the new category, and is taken to the transactions page for that category.
  - When the user clicks on the "Back" button (<), the user navigates to the transactions page for that category.

### Presentation video: https://www.loom.com/share/8b8ad2d15d6d4937bcf3daf7c9cd47fc

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,24 +1,71 @@
...
+# Budgetplus
 
-This README would normally document whatever steps are necessary to get the
-application up and running.
+This is a mobile web application where you can manage your budget: you have a list of transactions associated with a category, so that you can see how much money you spent and on what.
 
-Things you may want to cover:
+> |Homepage|Details page|
+> |-------------------------------------|-------------------------------------|
+> |![screen1](https://user-images.githubusercontent.com/9049260/154673640-28a66eff-be3e-4e85-93b7-60b1c9b3f747.png) | ![screen2](https://user-images.githubusercontent.com/9049260/154673687-4180322b-3da4-4b64-822b-533870b56c1b.png)|!
```
Comment (DeliceLydia) [inline README.md:7]: - [x] Please remove commented codes

File: README.md [hunk 2]
```diff
@@ -1,24 +1,71 @@
...
+## Prerequisites
+  - Ruby
+  - Rails
+  - PostgreSQL
 
-* Database creation
+## Live Demo on Heroku
 
-* Database initialization
+  - [Railsbugdetplus](https://railsbudgetplus.herokuapp.com/)
```
Comment (DeliceLydia) [inline README.md:21]: - [x] Fix your heroku link

**Review by DeliceLydia — CHANGES_REQUESTED** [review]
## Hi @ntzwilly, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!


## Required Changes ♻️
_Check the comments under the review._



### Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)

File: README.md (hunk 1 above)
Comment (ntzwilly) [reply README.md:7]: Hi @DeliceLydia, thank you for your feedback. Kindly note that Line 5 to 7 are not commented code. I use them to layout the screenshots of the application in the readme file

File: README.md (hunk 2 above)
Comment (ntzwilly) [reply README.md:21]: Thank you, upon checking, the link is working. Maybe I am doing it in the wrong way? But when I click on the "Railsbudgetplus" link in the readme file, I am redirected to the Heroku app.

File: README.md [hunk 3]
```diff
@@ -1,24 +1,71 @@
...
```
Comment (enaburekhan) [inline README.md:20]: - [x] I get the below error when I click on the Heroku link:

image
It will be nice to fix it.

**Review by enaburekhan — CHANGES_REQUESTED** [review]
Hi ntzwilly, 

Great work on making the changes 👏 
You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.


## Suggested changes
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._


Cheers and Happy coding!👏👏👏

Code review by **Eric Enaburekhan**

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

File: README.md (hunk 3 above)
Comment (ntzwilly) [reply README.md:20]: Hi @enaburekhan, could you please check at your end? The link is working on my side. I even tried it on 3 differents devices. Kindly check the screenshot below: 

Screenshot from 2022-02-18 20-08-03

Comment (enaburekhan) [reply README.md:20]: Hi @ntzwilly, I don't know why it's not working on my machine. I just tried again.

**Review by bimbolabuari — APPROVED** [review]
# Project Approved 🚀

Hi @ntzwilly 🥇,

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉



Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**